packages/hash/subgraph/src/stdlib/roots.ts

```diff
@@ -28,17 +25,17 @@ import { getPropertyTypeByEditionId } from "./element/property-type";
 export const getRoots = <RootType extends SubgraphRootType>(
   subgraph: Subgraph<RootType>,
 ): RootType["element"][] =>
-  subgraph.roots.map((rootEditionId) => {
+  subgraph.roots.map((rootVertexId) => {
     const root = mustBeDefined(
-      subgraph.vertices[rootEditionId.baseId]?.[
+      subgraph.vertices[rootVertexId.baseId]?.[
         // We could use type-guards here to convince TS that it's safe, but that would be slower, it's currently not
         // smart enough to realise this can produce a value of type `Vertex` as it struggles with discriminating
         // `EntityId` and `BaseUri`
         // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-        (rootEditionId as any).version
+        rootVertexId.version as any
```

Comment:
Swapping this around is interesting to me.. do you still need the `as any`?

Reply:
Yes, otherwise it results in 
> Element implicitly has an 'any' type because index expression is not of type 'number'